crates/ntx-builder/src/transaction.rs

```diff
@@ -91,7 +91,6 @@ impl NtxContext {
             .set_attribute("reference_block.number", chain_tip_header.block_num());
 
         async move {
-            // Work-around for `TransactionExecutor` not being `Send`.
             async move {
```

Comment:
Question: why do we need nested `async move`s here?

Reply:
For the inspection and span wrapping for futures, but the returned future would have incompatible lifetimes last time I checked